include/umf/memspaces/memspace_numa.h

```diff
@@ -0,0 +1,28 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#ifndef UMF_MEMSPACE_NUMA_H
+#define UMF_MEMSPACE_NUMA_H 1
+
+#include <umf/base.h>
+#include <umf/memspace.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum umf_result_t
+umfMemspaceCreateFromNumaId(size_t id,
+                            umf_memspace_handle_t *memspace /* out */);
+
```

Comment:
We should also add something to create a memspace from multiple NUMA nodes. One option is to pass an array of node ids but I think a better choice would be to have something like umfMemspaceCreateFromNumaMask() which would be more consistent with os_provider.

Reply:
Should we have just a single API to create memspace from a single node and multiple nodes? I mean that API that accepts NUMA-mask can be also used for a case with a single NUMA node. So we can remove thee the above function.

Reply:
Yes, that's also an option, but in C it's a bit cumbersome to set this mask so I felt that it might be useful to have a simpler API for this case. However, we could probably just implement some helpers for for managing those masks.

Reply:
If we want to make this user friendly we should add an array and mask versions. Note that the array version should also support a single NUMA node case so we don't need to provide a separate function for a single NUMA node. Please also note, that if we have a version for a single node then we have to also add functions to concatenate two memspaces and I don't think we want to introduce such API right now.

Reply:
Ok, I've exposed the version with array only for now. I can also create a PR to make os provider take array instead of mask as well. 

Once we finish the implementation for NUMA, we can add some example/test that will show how to use this with bitmasks. I think we can expose something like this:

```c++
void umfBitmaskToArray(unsigned long *nodemask, unsigned long maxnode, size_t *ids);
```